### ISSUE TITLE
Panic formatting value of type bytes read from ZNG stream

### ISSUE THREAD

**Issue by nwt** [brimdata/zed#2533]
On branch main at ca258e1:

```sh
$ zq - <<<{b:0x00} | zq -z -
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x155ead4]

goroutine 1 [running]:
github.com/brimdata/zed/zson.formatType(0xc00019ede0, 0xc00019ee00, 0x0, 0x0)
	/Users/noah/looky/zed/zson/formatter.go:493 +0x474
github.com/brimdata/zed/zson.formatType(0xc00019ede0, 0xc00019ee00, 0x1ad0ac8, 0xc0003cd530)
	/Users/noah/looky/zed/zson/formatter.go:457 +0x837
github.com/brimdata/zed/zson.FormatType(...)
	/Users/noah/looky/zed/zson/formatter.go:430
github.com/brimdata/zed/zson.(*Context).LookupTypeRecord(0xc0003ab040, 0xc0003acba0, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zson/context.go:99 +0x271
github.com/brimdata/zed/zio/zngio.(*Reader).readTypeRecord(0xc0003b77a0, 0x1, 0xc000420000)
	/Users/noah/looky/zed/zio/zngio/reader.go:337 +0x1f0
github.com/brimdata/zed/zio/zngio.(*Reader).readPayload(0xc0003b77a0, 0x0, 0x2900108, 0x40135f8, 0x58, 0x68)
	/Users/noah/looky/zed/zio/zngio/reader.go:155 +0xb6
github.com/brimdata/zed/zio/zngio.(*Reader).ReadPayload(0xc0003b77a0, 0x100f818, 0x68, 0x1901e80, 0x1903201)
	/Users/noah/looky/zed/zio/zngio/reader.go:109 +0x38
github.com/brimdata/zed/zio/zngio.(*Reader).Read(0xc0003b77a0, 0x1901e80, 0xc0003b77a0, 0x2e)
	/Users/noah/looky/zed/zio/zngio/reader.go:96 +0x2b
github.com/brimdata/zed/zio/detector.match(0x1ac1020, 0xc0003b77a0, 0x1938ce6, 0x3, 0x80000, 0xa00000)
	/Users/noah/looky/zed/zio/detector/reader.go:111 +0x38
github.com/brimdata/zed/zio/detector.NewReaderWithOpts(0x1ac0f40, 0xc0003ccf00, 0xc0003aad20, 0x194c292, 0xe, 0x1939233, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/noah/looky/zed/zio/detector/reader.go:80 +0xed4
github.com/brimdata/zed/zio/detector.OpenFromNamedReadCloser(0xc0003aad20, 0x402df18, 0xc000010010, 0x194c292, 0xe, 0x1939233, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/noah/looky/zed/zio/detector/file.go:44 +0x145
github.com/brimdata/zed/zio/detector.OpenFileWithContext(0x1ad0010, 0xc0000380c8, 0xc0003aad20, 0x194c292, 0xe, 0x1939233, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/noah/looky/zed/zio/detector/file.go:33 +0x238
github.com/brimdata/zed/zio/detector.OpenFile(...)
	/Users/noah/looky/zed/zio/detector/file.go:21
github.com/brimdata/zed/cli/inputflags.(*Flags).Open(0xc00021b200, 0xc0003aad20, 0xc0000321d0, 0x1, 0x1, 0x1, 0x0, 0x203000, 0x203000, 0x1ebf5a0, ...)
	/Users/noah/looky/zed/cli/inputflags/flags.go:91 +0x178
github.com/brimdata/zed/cmd/zed/query.(*Command).Run(0xc00021b1e0, 0xc0000321d0, 0x1, 0x1, 0x0, 0x0)
	/Users/noah/looky/zed/cmd/zed/query/command.go:162 +0x5ce
github.com/brimdata/zed/pkg/charm.(*instance).run(0xc00008b4c0, 0xc0000321c0, 0x2, 0x2, 0x0, 0x0)
	/Users/noah/looky/zed/pkg/charm/instance.go:56 +0x2a2
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0x1f32180, 0xc0000321c0, 0x2, 0x2, 0x1, 0xc000010150, 0x0, 0x1)
	/Users/noah/looky/zed/pkg/charm/charm.go:77 +0x85
main.main()
	/Users/noah/looky/zed/cmd/zq/main.go:13 +0xe5
exit status 2
```

**Comment by nwt** [comment]
Fixed in #2529.

**Comment by philrz** [comment]
Verified in `zq` commit 79c4047. No more panic.

```
$ zq -version
Version: v0.29.0-240-g79c40474

$ zq - <<<{b:0x00} | zq -z -
{b:0x00}
```

Thanks @nwt!
